openssl: add versions 1.1.1u/3.0.9/3.1.1

gegles opened the pull request:
Specify library name and version:  **openssl/1.1.1u|3.0.9|3.1.1**

Routine upgrade to add latest security patch updates.

<!-- This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks! -->


---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Croydon commented on recipes/openssl/1.x.x/conandata.yml:
```diff
@@ -26,6 +26,11 @@ sources:
+  1.1.1u:
+    sha256: e2f8d84b523eecd06c7be7626830370300fbcc15386bf5142d72758f6963ebc6
+    url:
+      - "https://www.openssl.org/source/openssl-1.1.1u.tar.gz"
```
```suggestion
      - "https://www.openssl.org/source/openssl-1.1.1u.tar.gz"
      - "https://www.openssl.org/source/old/1.1.1/openssl-1.1.1u.tar.gz"
```

gegles replied:
@Croydon, I had tried it with that URL, but it gives me a 404:
```
wget https://www.openssl.org/source/old/1.1.1/openssl-1.1.1u.tar.gz
--2023-05-30 13:41:57--  https://www.openssl.org/source/old/1.1.1/openssl-1.1.1u.tar.gz
Resolving www.openssl.org (www.openssl.org)... 96.16.16.41
Connecting to www.openssl.org (www.openssl.org)|96.16.16.41|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-05-30 13:41:59 ERROR 404: Not Found.
```

gegles replied:
Looking at [this](https://www.openssl.org/source/old/1.1.1/) URL, I can see that `t` and `u` are not (yet) in the old folder.
If the idea is the futureproof this so that these mirrors will eventually work, I am happy to add them back in the recipe...

Croydon replied:
@gegles That is the idea, yes

gegles commented:
Ok @Croydon, I think I understand how this old folder is used eventually.. So I've now added the old mirrors for 1.1.1u and also went back and added all the old mirrors for 3.x.x.. Hope it's ok. thx!

Croydon commented:
Please undo for 3.x
The OpenSSL project eventually recognized the need for permanent valid URLs, so they won't move the releases for 3.x anymore

gegles commented:
> Please undo for 3.x The OpenSSL project eventually recognized the need for permanent valid URLs, so they won't move the releases for 3.x anymore

Ok, done. thx!

Nekto89 commented on recipes/openssl/config.yml:
```diff
@@ -1,7 +1,11 @@
 versions:
   # 3.0.x releases
+  3.1.1:
+    folder: "3.x.x"
   3.1.0:
     folder: "3.x.x"
+  3.0.9:
```
I'm not really sure that comments are useful, but at least they shouldn't lie

RubenRBS replied:
Good catch!

gegles commented:
@RubenRBS @Croydon looks like this ready to merged. Thanks!

RubenRBS commented:
We're using this recipe bump as a test that we haven't glossed over anything for moving forward with version ranges, sorry about the delay getting this merged :)
